### PR TITLE
Add sentry-native 0.4.14

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **sentry-native/0.4.14**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sentry-native/all/conanfile.py [hunk 1]
```diff
@@ -130,6 +130,8 @@ def validate(self):
             raise ConanInvalidConfiguration("This version doesn't support ARM compilation")
 
     def build_requirements(self):
+        if tools.Version(self.version) >= "0.4.1":
```
Comment (SpaceIm) [inline recipes/sentry-native/all/conanfile.py:133]:
```suggestion
        if self.settings.os == "Windows" and tools.Version(self.version) >= "0.4.0":
```

Comment (SpaceIm) [reply recipes/sentry-native/all/conanfile.py:133]: https://github.com/getsentry/sentry-native/blob/0.4.0/CMakeLists.txt#L1-L8

Comment (SpaceIm) [reply recipes/sentry-native/all/conanfile.py:133]: Honestly, it should be replaced upstream by:

```cmake
cmake_minimum_required(VERSION 3.10)
if(POLICY CMP0091)
    cmake_policy(SET CMP0091 NEW)
endif()
```

Comment (MartinDelille) [reply recipes/sentry-native/all/conanfile.py]: I have no idea what the issue is with 0.4.1... Is it an option to drop support for old version ? This PR aims only to add add the latest version to the CCI.

**Comment by MartinDelille** [comment]
Closing in favor of #9445

**Pull request closed without merging**